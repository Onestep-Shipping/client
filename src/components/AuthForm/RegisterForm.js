/* eslint-disable no-unused-vars */
import React, {useState, useCallback} from 'react';
import './AuthForm.css';

import Input from '../Input/Input';
import app from '../../firebase/base.js';
import PropTypes from 'prop-types';

const RegisterForm = (props) => {
  const { closeModal } = props;

  const [errors, setErrors] = useState([]);
  const [displayErrors, setDisplayErrors] = useState(false);


  const handleSignUp = useCallback(async (e) => {
    e.preventDefault();
    const email = e.target[6].value;
    const password = e.target[7].value;
    if (email.length > 0 && password.length > 0) {
      try {
        await app
            .auth()
            .createUserWithEmailAndPassword(email, password);
        closeModal();
      } catch (error) {
        alert(error);
      }
    }
  }, [closeModal]);

  return (
    <div className="inner-container">
      <div className="header">Register</div>
      <form className="register-box" onSubmit={handleSignUp} noValidate>
        {errors.length > 0 &&
          errors.map((err, ind) => <small className="danger-error" key={ind}>{err}</small>)
        }
        <div className="all-texfield-register-container">
          <div className="company-info-container">
            <Input name="Company Name" type="text" displayErrors={displayErrors} />
            <Input name="Address" type="address" 
              displayErrors={displayErrors} 
              placeholder="Street"
            />
            <Input name="" type="address" 
              displayErrors={displayErrors} 
              placeholder="City, Province & Postal Code"
            />
            <Input name="" type="address" 
              displayErrors={displayErrors} 
              placeholder="Country"
            />
            <Input name="Telephone" type="phone" displayErrors={displayErrors} />
          </div>
          <div>
            <Input name="Tax ID" type="text" displayErrors={displayErrors} />
            <Input name="Person In-Charge" type="text" displayErrors={displayErrors} />
            <Input name="Position" type="text" displayErrors={displayErrors} />
            <Input name="Email" type="email" displayErrors={displayErrors} />
            <Input name="Password" type="password" displayErrors={displayErrors} />
          </div>
        </div>
        <button className="login-btn">Sign Up</button>
      </form>
    </div>
  );
};

export default RegisterForm;

RegisterForm.propTypes = {
  closeModal: PropTypes.func,
};
