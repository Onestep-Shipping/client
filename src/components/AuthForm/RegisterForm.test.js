import React from 'react';
import RegisterForm from './RegisterForm';
import { shallow } from 'enzyme';

const closeModal = () => {}

let wrapped = shallow( <RegisterForm closeModal={closeModal} />);

describe('RegisterForm', () => {
  it('should render the RegisterForm Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
});