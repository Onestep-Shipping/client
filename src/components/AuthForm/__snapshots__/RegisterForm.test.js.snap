// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render the RegisterForm Component correctly 1`] = `
<div
  className="inner-container"
>
  <div
    className="header"
  >
    <span
      className="schedule-header-text"
    >
      Register
    </span>
  </div>
  <form
    className="register-box"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="all-texfield-register-container"
    >
      <div
        className="company-info-container"
      >
        <Input
          displayErrors={false}
          name="Company Name"
          type="text"
        />
        <Input
          displayErrors={false}
          name="Address"
          placeholder="Street"
          type="address"
        />
        <Input
          displayErrors={false}
          name=""
          placeholder="City, Province & Postal Code"
          type="address"
        />
        <Input
          displayErrors={false}
          name=""
          placeholder="Country"
          type="address"
        />
        <Input
          displayErrors={false}
          name="Telephone"
          type="phone"
        />
      </div>
      <div>
        <Input
          displayErrors={false}
          name="Tax ID"
          type="text"
        />
        <Input
          displayErrors={false}
          name="Person In-Charge"
          type="text"
        />
        <Input
          displayErrors={false}
          name="Position"
          type="text"
        />
        <Input
          displayErrors={false}
          name="Email"
          type="email"
        />
        <Input
          displayErrors={false}
          name="Password"
          type="password"
        />
      </div>
    </div>
    <button
      className="login-btn"
    >
      Sign Up
    </button>
  </form>
</div>
`;
